saq/queue/postgres.py

```diff
@@ -459,19 +459,16 @@ async def listen(
     async def notify(self, job: Job, connection: AsyncConnection | None = None) -> None:
         await self._notify(job.key, job.status, connection)
 
-    async def update(
-        self,
-        job: Job,
-        connection: AsyncConnection | None = None,
-        expire_at: float | None = -1,
-        **kwargs: t.Any,
-    ) -> None:
-        job.touched = now()
-
-        for k, v in kwargs.items():
-            setattr(job, k, v)
+    async def _update(self, job: Job, status: Status | None = None, **kwargs: t.Any) -> None:
+        expire_at = kwargs.pop("expire_at", -1)
+        connection = kwargs.pop("connection", None)
 
         async with self.nullcontext(connection) if connection else self.pool.connection() as conn:
+            if not status:
+                status = await self.get_job_status(job.key, for_update=True, connection=conn)
```

Comment:
I think there's still a race condition here, if the job is aborted after this call but before the update. So this will still happen (just less frequently).

Not sure if you care about that.

Probably need to push the partial update all the way down to the db update.

Reply:
isn't that what for_update accomplishes?

Reply:
i can also use a transaction if necessary

Reply:
uses a transaction now, there should be no race condition

Reply:
ah, i didn't realize we select for update

Reply:
(i havent been following postgres queue closely)

Reply:
you should the redis queue is falling out of fashion… no transactions lol

Reply:
airbnb doesn't host postgres, so not a chance haha